### PR TITLE
Controls

### PR DESCRIPTION
**Pull request by ThatIsAUsername**
Functional, but not super pretty. I may have violated some best practices. Actually, I should wait for you to tell me that.
Probably need to save/load settings to a config file before this hits real users.

### CODE REVIEW

File: src/UI/Art/SpriteArtist/ControlOptionsSetupArtist.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+   */
+  private static class KeyBindingPanel
+  {
+    InputHandler.InputAction myAction;
+
+    // A couple of helper quantities.
+    private static final int textBufferPx = 4;
+    private static final int keyBufferPx = 8;
+
+    // The composed TeamPanel image.
```
Comment (ThislsAUsername) [inline src/UI/Art/SpriteArtist/ControlOptionsSetupArtist.java:84]: A team of keybindings? :stuck_out_tongue:

File: src/UI/InputHandler.java [hunk 2]
```diff
@@ -3,11 +3,16 @@
 import java.awt.event.KeyEvent;
 import java.util.ArrayList;
 import java.util.Arrays;
-public class InputHandler
+import java.util.HashMap;
+
+import Engine.IController;
+import Engine.OptionSelector;
+import UI.InputHandler.InputAction;
+public class InputHandler implements IController
```
Comment (ThislsAUsername) [inline src/UI/InputHandler.java:17]: Hmm. I can't say I expected to see this, but It does make a certain amount of sense as a way to keep input-specifics in this one class to abstract it away from the rest of the codebase.

**Comment by ThislsAUsername** [comment]
Some general UI feedback, screenshot for reference
image
It'd be nice to x-align the first key panel for each input item.
The fact that the action types have up-right shadows and `ADD` has down-left shadows looks a little weird and inconsistent to me. A good alternative, I think, would be to just make both light (or one light/one dark) to distinguish them.

File: src/UI/Art/SpriteArtist/ControlOptionsSetupArtist.java (hunk 1 above)
Comment (ThatIsAUsername) [reply src/UI/Art/SpriteArtist/ControlOptionsSetupArtist.java:84]: Of course. A single bound key wouldn't be worth a whole lot in this game.

File: src/UI/InputHandler.java (hunk 2 above)
Comment (ThatIsAUsername) [reply src/UI/InputHandler.java:17]: Well, I didn't expect to do this, but then it kinda happened.

**Comment by ThatIsAUsername** [comment]
Yeah, I did the up-right shadow on the left column because the vertical line on the right provides some separation and differentiation for those panels. It does look a tad wonky though.

**Comment by ThislsAUsername** [comment]
Yeah. That looks good to me.

File: src/UI/InputHandler.java (hunk 2 above)
Comment (ThislsAUsername) [reply src/UI/InputHandler.java:17]: Hmm. I feel like the static and non-static portions aren't particularly closely-bound.
It is amusing/novel to leave them together, though, so I guess I don't care much either way.

**Comment by ThislsAUsername** [comment]
Works for me. Did you want to make any other changes?

File: src/UI/InputHandler.java (hunk 2 above)
Comment (ThatIsAUsername) [reply src/UI/InputHandler.java:17]: I started out by making a separate class, but it did nothing but make calls to InputHandler anyway.
I imagine I could find a more structured way to compose it if I tried, but it's simple enough and works well enough now that it doesn't seem worth the effort to try and overthink it at this point.

**Comment by ThatIsAUsername** [comment]
There are any number of changes that _could_ be done, like putting in a "restore defaults" button somewhere, or changing the sub-menu on the main page to not stand out so much (since I'm still not a huge fan of that), but this is quite usable, so I think I'm fine with it.

**Pull request merged**